src/common/scripts/modules/Block.ts

```diff
@@ -0,0 +1,166 @@
+import EventBus from './EventBus';
+
+export default class Block {
+    static EVENTS = {
+        INIT: 'init',
+        FLOW_CDM: 'flow:component-did-mount',
+        FLOW_CDU: 'flow:component-did-update',
+        FLOW_RENDER: 'flow:render'
+    };
+   
+    _element: HTMLElement = document.createElement('template');
+    _meta: {
+        tagName: string
+        props: Record<string, any>
+    }
+    _handlers: Record<string, () => void> = {};
+    props: Record<string, any>;
+    eventBus: () => EventBus;
+   
+    /** JSDoc
+     * @param {string} tagName
+     * @param {Object} props
+     *
+     * @returns {void}
+     */
+    constructor(tagName = 'div', props: Record<string, any> = {}) {
+        const eventBus = new EventBus();
+        this._meta = {
+            tagName,
+            props
+        };
+   
+        this.props = this._makePropsProxy(props);
+    
+        this.eventBus = () => eventBus;
+    
+        this._registerEvents(eventBus);
+        eventBus.emit(Block.EVENTS.INIT);
+    }
+
+    _addHandlers(): void {
+        const { handlers = {} } = this.props;
+        this._handlers = handlers;
+
+        Object.keys(handlers).forEach((handler) => {
+            this._element.addEventListener(handler, handlers[handler])
```

Comment:
```suggestion
            this._element.addEventListener(handler, handlers[handler]);
```
Вообще по стилю написания кода стоит пробежаться и привести к одному виду